tar/util.c

```diff
@@ -140,6 +140,7 @@ safe_fprintf(FILE *f, const char *fmt, ...)
 		} else {
 			/* Leave fmtbuff pointing to the truncated
 			 * string in fmtbuff_stack. */
+			fmtbuff = fmtbuff_stack;
```

Comment:
I would agree in theory, but if a malloc call fails on line 132, it will dereference a NULL pointer instead of trying to fall back to using the stack again.

It might be better to just try using the heap memory first, then fall back to the truncated stack memory for clarity and conciseness.